include/DoocsBackendNumericRegisterAccessor.h

```diff
@@ -0,0 +1,371 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+#pragma once
+
+#include "DoocsBackendRegisterAccessor.h"
+#include <eq_client.h>
+#include <type_traits>
+
+#include <ChimeraTK/SupportedUserTypes.h>
+
+#include <string>
+
+namespace ChimeraTK {
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  class DoocsBackendNumericRegisterAccessor : public DoocsBackendRegisterAccessor<UserType> {
+   public:
+    ~DoocsBackendNumericRegisterAccessor() override;
+
+   protected:
+    DoocsBackendNumericRegisterAccessor(boost::shared_ptr<DoocsBackend> backend, const std::string& path,
+        const std::string& registerPathName, size_t numberOfWords, size_t wordOffsetInRegister, AccessModeFlags flags);
+
+    void doPostRead(TransferType type, bool hasNewData) override;
+
+    void doPreWrite(TransferType type, VersionNumber) override;
+
+    // simple helper function to call the correct EqData::get_...() function to fetch data from dst
+    template<typename T>
+    T dataGet(int index);
+
+    // Helper function to call a callable for a C++ numeric type corresponding to the type in the EqData object.
+    // Calling this function for an unsupported data type will throw a runtime error.
+    // Note: This function does not support IFFF, since the data type depends on the index there. Calling this
+    // function with an IFFF raises an assertion.
+    template<typename CALLABLE>
+    void callForDoocsType(const EqData& data, CALLABLE callable);
+
+    friend class DoocsBackend;
+  };
+
+  /********************************************************************************************************************/
+
+  template<typename UserType>
+  DoocsBackendNumericRegisterAccessor<UserType>::DoocsBackendNumericRegisterAccessor(
+      boost::shared_ptr<DoocsBackend> backend, const std::string& path, const std::string& registerPathName,
+      size_t numberOfWords, size_t wordOffsetInRegister, AccessModeFlags flags)
+  : DoocsBackendRegisterAccessor<UserType>(
+        backend, path, registerPathName, numberOfWords, wordOffsetInRegister, flags) {
+    // check data type
+    // Note: We cannot rely subsequently that the data type remains unchanged, since it might change either dynamically
+    // or by replacing the server (the data type might even come from the cache file).
+    try {
+      // callForDoocsType() will throw a runtime_error if the type is not supported, except for DATA_IFFF which we have
+      // to catch first
+      if(this->src.type() != DATA_IFFF) {
+        callForDoocsType(this->src, [](auto) {});
+      }
+    }
+    catch(ChimeraTK::runtime_error&) {
+      // This is not a real runtime_error here, since this should be already prevented by logic in the backend
+      assert(false);
+      this->shutdown();
+      std::terminate();
+    }
```

Comment:
This is basically an assertion to make sure this accessor type is not used for the wrong data types. Before there was a big if block checking for all types, but I wanted to avoid another place where all types need to be listed, hence this trick with `callForDoocsType()` and the exception.

Reply:
Maybe put a comment. Code is OK